### PR TITLE
AWS connectors: align client setup docs

### PR DESCRIPTION
**Pull request by ennru**
## Purpose

Align the docs how the AWS SDK clients are created using Akka HTTP as an underlying HTTP client.

## References

References #2014 

## Changes

* new docs page "AWS retry configuration"
* Use Akka HTTP even for AWS Lambda and Kinesis
* Show Akka HTTP client setup in all examples
* Show more imports in examples
* Use Akka 2.6.0 for cross builds

### CODE REVIEW

**Comment by ennru** [comment]
@gabfssilva @regiskuckaertz could you have a look and check if I got it right?

**Comment by gabfssilva** [comment]
> @gabfssilva @regiskuckaertz could you have a look and check if I got it right?

LGTM

File: project/Dependencies.scala [hunk 1]
```diff
@@ -296,7 +310,7 @@ object Dependencies {
           ExclusionRule("software.amazon.awssdk", "netty-nio-client")
         ),
         "org.mockito" % "mockito-core" % mockitoVersion % Test // MIT
-      )
+      ) ++ JacksonDatabindDependencies
```
Comment (regiskuckaertz) [inline project/Dependencies.scala]: woops, good catch 🤭

**Comment by regiskuckaertz** [comment]
👍 Looks great!

File: awslambda/src/test/scala/docs/scaladsl/Examples.scala [hunk 2]
```diff
@@ -20,20 +17,37 @@ object Examples {
...
-    .build()
-  Source.single(request).via(AwsLambdaFlow(1)).runWith(Sink.seq)
-  //#run
+  def initClient(): Unit = {
+    //#init-client
+    import com.github.matsluni.akkahttpspi.AkkaHttpClient
+    import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+    import software.amazon.awssdk.services.lambda.LambdaAsyncClient
+
+    val credentials = AwsBasicCredentials.create("x", "x")
```
Comment (seglo) [inline awslambda/src/test/scala/docs/scaladsl/Examples.scala]: [minor] I know isn't new code, but maybe we should add a comment link to best practices for AWS key management so users don't just copy and paste this and hardcode credentials.  I found this resource: https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html

File: docs/src/main/paradox/awslambda.md [hunk 3]
```diff
@@ -18,16 +18,22 @@ The table below shows direct dependencies of this module and the second tab show
...
-Flow provided by this connector needs a prepared `LambdaAsyncClient` to be able to invoke lambda functions.
+Flow provided by this connector needs a prepared @javadoc[LambdaAsyncClient](software.amazon.awssdk.services.lambda.LambdaAsyncClient) to be able to invoke lambda functions.
 
 Scala
 : @@snip (/awslambda/src/test/scala/docs/scaladsl/Examples.scala) { #init-client }
 
 Java
 : @@snip (/awslambda/src/test/java/docs/javadsl/Examples.java) { #init-client }
 
+This connector is set up to use @extref:[Akka HTTP](akka-http:) as default HTTP client via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused, if not set explicitly a separate actor system will be created internally.
```
Comment (seglo) [inline docs/src/main/paradox/awslambda.md]:
```suggestion
This connector is setup to use @extref:[Akka HTTP](akka-http:) as the default HTTP client implementation via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused.  If it is not set explicitly then a separate actor system will be created internally.
```

File: docs/src/main/paradox/kinesis.md [hunk 4]
```diff
@@ -163,6 +169,12 @@ Scala
 Java
 : @@snip [snip](/kinesis/src/test/java/docs/javadsl/KinesisFirehoseSnippets.java) { #init-client }
 
+This connector is set up to use @extref:[Akka HTTP](akka-http:) as default HTTP client via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused, if not set explicitly a separate actor system will be created internally.
```
Comment (seglo) [inline docs/src/main/paradox/kinesis.md]:
```suggestion
This connector is setup to use @extref:[Akka HTTP](akka-http:) as the default HTTP client implementation via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused.  If it is not set explicitly then a separate actor system will be created internally.
```

File: docs/src/main/paradox/sns.md [hunk 5]
```diff
@@ -31,7 +31,9 @@ Scala
 Java
 : @@snip [snip](/sns/src/test/java/docs/javadsl/SnsPublisherTest.java) { #init-client }
 
-Alpakka SQS and SNS are set up to use @extref:[Akka HTTP](akka-http:) as default HTTP client via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused, if not set explicitly a separate actor system will be created internally.
+This connector is set up to use @extref:[Akka HTTP](akka-http:) as default HTTP client via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused, if not set explicitly a separate actor system will be created internally.
```
Comment (seglo) [inline docs/src/main/paradox/sns.md]:
```suggestion
This connector is setup to use @extref:[Akka HTTP](akka-http:) as the default HTTP client implementation via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused.  If it is not set explicitly then a separate actor system will be created internally.
```

File: docs/src/main/paradox/sqs.md [hunk 6]
```diff
@@ -47,9 +47,11 @@ Scala
 Java
 : @@snip [snip](/sqs/src/test/java/akka/stream/alpakka/sqs/javadsl/BaseSqsTest.java) { #init-client }
 
-### Underlying HTTP client
+This connector is set up to use @extref:[Akka HTTP](akka-http:) as default HTTP client via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused, if not set explicitly a separate actor system will be created internally.
```
Comment (seglo) [inline docs/src/main/paradox/sqs.md]:
```suggestion
This connector is setup to use @extref:[Akka HTTP](akka-http:) as the default HTTP client implementation via the thin adapter library [AWS Akka-Http SPI implementation](https://github.com/matsluni/aws-spi-akka-http). By setting the `httpClient` explicitly (as above) the Akka actor system is reused.  If it is not set explicitly then a separate actor system will be created internally.
```

**Comment by ennru** [comment]
Good point about the credentials, @seglo 
I pushed this further and made it a "Shared AWS client configuraion" doc page instead.

**Pull request merged**